### PR TITLE
chore: Move plugin docs to website

### PR DESCRIPTION
**Pull request by yevgenypats**
We outgrew our plugin documentation (which is good) were it was hosted directly on GitHub. Now that we have a monorepo it is actually quite easy to just write the plugin docs directly in the Nextra website. This will have the following benefits:
 
1) Content hosted on our domain
2) Searchable with Nextra
3) We wont need the search/replace step in all markdown files every single release because we can use React to import latest versions. This is now solved with an improvement I made that enabled variable expansion inside code snippets of `shiki` library

Caveat: The only thing that will still be hosted on GitHub is the tables due to the huge size of those docs and Nextra doesn't deal well with that at the moment (might move that in the future).

### CODE REVIEW

**Comment by yevgenypats** [comment]
Broken links should be fixed post merge

**Comment by hermanschaaf** [comment]
@yevgenypats Just clicking around on the website preview, it looks like the "Plugins" link at the top of the page no longer works: 

Screenshot 2022-11-28 at 08 58 28

Screenshot 2022-11-28 at 08 58 32

File: website/pages/docs/plugins/destinations/postgresql/overview.md
Comment (hermanschaaf) [inline website/pages/docs/plugins/destinations/postgresql/overview.md:5]:
```suggestion
Supported database versions:
```

(nit, just noticed it while reading through)

File: website/components/mdx/_download-macOS.mdx
Comment (hermanschaaf) [inline website/components/mdx/_download-macOS.mdx:27]: Why is this link using `${VERSION_CLI}` while the one higher up is using just plain `VERSION_CLI`? Is there a difference?

File: website/next.config.mjs
Comment (hermanschaaf) [inline website/next.config.mjs:5]:
```suggestion
const reSourcePluginVersion = /VERSION_SOURCE_([a-zA-Z0-9_]+)/;
const reDestPluginVersion = /VERSION_DESTINATION_([a-zA-Z0-9_]+)/;
```

(also a nit)

File: website/components/mdx/_download-macOS.mdx
Comment (hermanschaaf) [reply website/components/mdx/_download-macOS.mdx:27]: Oh yeah no, looking at the preview site, looks like this is a bug
Screenshot 2022-11-28 at 09 17 18

File: website/pages/docs/plugins/sources/k8s/configuration.md
Comment (hermanschaaf) [inline website/pages/docs/plugins/sources/k8s/configuration.md:13]:
```suggestion
  version: "VERSION_SOURCE_K8S"
```

File: website/next.config.mjs
Comment (hermanschaaf) [inline website/next.config.mjs:78]: Wow, nice find that you can do this!!

File: website/pages/docs/plugins/sources/terraform/_meta.json
Comment (hermanschaaf) [inline website/pages/docs/plugins/sources/terraform/_meta.json:4]:
```suggestion
    "title": "Tables ↗",
```

I think it would good to indicate on all the Tables links that the links open in a new window by adding a little unicode arrow.

I'd like to bring the list of tables onto the site as well (not necessarily the columns, though), but we can do that in a follow-up.

File: website/pages/docs/plugins/sources/aws/_meta.json
Comment (hermanschaaf) [inline website/pages/docs/plugins/sources/aws/_meta.json:6]:
```suggestion
    "title": "Tables ↗",
```

File: website/pages/docs/plugins/sources/azure/_meta.json
Comment (hermanschaaf) [inline website/pages/docs/plugins/sources/azure/_meta.json:5]:
```suggestion
    "title": "Tables ↗",
```

File: website/pages/docs/plugins/sources/cloudflare/_meta.json
Comment (hermanschaaf) [inline website/pages/docs/plugins/sources/cloudflare/_meta.json:5]:
```suggestion
    "title": "Tables ↗",
```

File: website/pages/docs/plugins/sources/digitalocean/_meta.json
Comment (hermanschaaf) [inline website/pages/docs/plugins/sources/digitalocean/_meta.json:4]:
```suggestion
    "title": "Tables ↗",
```

File: website/pages/docs/plugins/sources/gcp/_meta.json
Comment (hermanschaaf) [inline website/pages/docs/plugins/sources/gcp/_meta.json:5]:
```suggestion
    "title": "Tables ↗",
```

File: website/pages/docs/plugins/sources/github/_meta.json
Comment (hermanschaaf) [inline website/pages/docs/plugins/sources/github/_meta.json:4]:
```suggestion
    "title": "Tables ↗",
```

File: website/pages/docs/plugins/sources/heroku/_meta.json
Comment (hermanschaaf) [inline website/pages/docs/plugins/sources/heroku/_meta.json:5]:
```suggestion
    "title": "Tables ↗",
```

File: website/pages/docs/plugins/sources/k8s/_meta.json
Comment (hermanschaaf) [inline website/pages/docs/plugins/sources/k8s/_meta.json:5]:
```suggestion
    "title": "Tables ↗",
```

File: website/pages/docs/plugins/sources/okta/_meta.json
Comment (hermanschaaf) [inline website/pages/docs/plugins/sources/okta/_meta.json:4]:
```suggestion
    "title": "Tables ↗",
```

**Review by hermanschaaf — CHANGES_REQUESTED** [review]
Amazing! 🎉 Just commented on a few things to look into before merging

File: website/components/mdx/_download-macOS.mdx
Comment (hermanschaaf) [inline website/components/mdx/_download-macOS.mdx:27]:
```suggestion
curl -L https://github.com/cloudquery/cloudquery/releases/download/VERSION_CLI/cloudquery_darwin_arm64 -o cloudquery
```

**Comment by yevgenypats** [comment]
@hermanschaaf fixed. PTAL :)

**Comment by hermanschaaf** [comment]
@yevgenypats Would it work to rename the `overview` pages to `index` (and update `meta.json`) so that the URL paths stay the same and external links don't get 404s?  

E.g. before it was `/docs/plugins`, now it is `/docs/plugins/overview`. And before it was `/docs/plugins/sources`, but now it's `/docs/plugins/sources/overview`

File: plugins/destination/csv/README.md
Comment (erezrokah) [inline plugins/destination/csv/README.md:7]:
```suggestion
- [User Guide](https://cloudquery.io/docs/plugins/destinations/csv/overview)
```

We shouldn't be using the old `docs` domain

File: plugins/destination/postgresql/README.md
Comment (erezrokah) [inline plugins/destination/postgresql/README.md:7]:
```suggestion
- [User Guide](https://cloudquery.io/docs/plugins/destinations/postgresql/overview)
```

File: plugins/destination/sqlite/README.md
Comment (erezrokah) [inline plugins/destination/sqlite/README.md:7]:
```suggestion
- [User Guide](https://cloudquery.io/docs/plugins/destinations/sqlite/overview)
```

File: plugins/source/aws/README.md
Comment (erezrokah) [inline plugins/source/aws/README.md:8]:
```suggestion
- [User Guide](https://cloudquery.io/docs/plugins/sources/aws/overview)
```

File: plugins/destination/sqlite/README.md
Comment (erezrokah) [inline plugins/destination/sqlite/README.md:7]:
```suggestion
- [User Guide](https://cloudquery.io/docs/plugins/destinations/sqlite/overview)
```

File: plugins/source/azure/README.md
Comment (erezrokah) [inline plugins/source/azure/README.md:7]:
```suggestion
- [User Guide](https://cloudquery.io/docs/plugins/sources/azure/overview)
```

File: plugins/source/aws/docs/configuration.md
Comment (erezrokah) [inline plugins/source/aws/docs/configuration.md:3]:
```suggestion
Moved to [https://cloudquery.io/docs/plugins/sources/aws/configuration](https://cloudquery.io/docs/plugins/sources/aws/configuration)
```

File: plugins/source/azure/docs/configuration.md
Comment (erezrokah) [inline plugins/source/azure/docs/configuration.md:3]:
```suggestion
Moved to [https://cloudquery.io/docs/plugins/sources/azure/configuration](https://cloudquery.io/docs/plugins/sources/azure/configuration)
```

File: plugins/source/digitalocean/README.md
Comment (erezrokah) [inline plugins/source/digitalocean/README.md:7]:
```suggestion
- [User Guide](https://cloudquery.io/docs/plugins/sources/digitalocean/overview)
```

File: plugins/source/heroku/docs/configuration.md
Comment (erezrokah) [inline plugins/source/heroku/docs/configuration.md:3]:
```suggestion
Moved to [https://cloudquery.io/docs/plugins/sources/heroku/configuration](https://cloudquery.io/docs/plugins/sources/heroku/configuration)
```

File: plugins/source/k8s/README.md
Comment (erezrokah) [inline plugins/source/k8s/README.md:7]:
```suggestion
- [User Guide](https://cloudquery.io/docs/plugins/sources/k8s/overview)
```

File: website/components/mdx/_download-linux.mdx
Comment (erezrokah) [inline website/components/mdx/_download-linux.mdx:8]: Maybe we should keep the latest link too? Otherwise there's no easy way to download CloudQuery latest version (without looking in the docs)

File: website/pages/_meta.json
Comment (erezrokah) [inline website/pages/_meta.json:24]: We should add a redirect in `vercel.json` to avoid the 404 this creates

File: website/next.config.mjs
Comment (erezrokah) [inline website/next.config.mjs:35]: 🚀

File: website/pages/docs/plugins/destinations/overview.mdx
Comment (erezrokah) [inline website/pages/docs/plugins/destinations/overview.mdx:5]: I think destinations are missing a `_meta.json` file, see:
https://cloudquery-fanvd8nsu-cloudquery.vercel.app/docs/plugins/destinations/overview
image

File: website/pages/docs/plugins/sources/_meta.json
Comment (erezrokah) [inline website/pages/docs/plugins/sources/_meta.json:2]: We should add a similar file under destinations

File: website/pages/docs/plugins/sources/overview.mdx
Comment (erezrokah) [inline website/pages/docs/plugins/sources/overview.mdx:5]: We should update the util in `versions` to reference the new location of versions under website, here:
https://github.com/cloudquery/cloudquery/blob/2ceb3214cd504a134f9087e8da5a167e9b11e8e5/website/utils/versions.ts#L4

**Review by erezrokah — CHANGES_REQUESTED** [review]
Great work @yevgenypats I like the new approach. Added a few comments

File: .github/workflows/release_cli.yml
Comment (yevgenypats) [reply .github/workflows/release_cli.yml:83]: Yeah let's remove it from github action as well and make it mandatory.

**Comment by hermanschaaf** [comment]
@yevgenypats Yeah it does some weird things, let's put a redirect in vercel.json instead like @erezrokah suggested :)

File: website/next.config.mjs
Comment (erezrokah) [inline website/next.config.mjs:31]:
```suggestion
function getVersions() {
  const files = fs
    .readdirSync("./versions", { withFileTypes: true })
    .filter((dirent) => dirent.isFile())
    .map((file) => ({
      name: path.basename(file.name, path.extname(file.name)),
      latest: JSON.parse(fs.readFileSync(`./versions/${file.name}`, "utf8"))
        .latest,
    }));

  const cli = files.find((file) => file.name === "cli").latest.split("-")[1];
  const sources = Object.fromEntries(
    files
      .filter((file) => file.name.startsWith("source-"))
      .map((file) => [file.name.split("-")[1], file.latest.split("-")[3]])
  );
  const destinations = Object.fromEntries(
    files
      .filter((file) => file.name.startsWith("destination-"))
      .map((file) => [file.name.split("-")[1], file.latest.split("-")[3]])
  );

  return { cli, sources, destinations };
}
```

[Non blocking] I think using `readdirSync` instead of a `while` loop makes this code easier to read

File: .github/workflows/release_cli.yml
Comment (erezrokah) [reply .github/workflows/release_cli.yml:83]: Opened https://github.com/cloudquery/setup-cloudquery/issues/143

File: website/next.config.mjs
Comment (yevgenypats) [reply website/next.config.mjs:31]: let's do it in a follow-up PR as already have a fatigue from this one and hard to test various changes

File: plugins/source/aws/docs/configuration.md
Comment (erezrokah) [inline plugins/source/aws/docs/configuration.md:3]:
```suggestion
Moved to [https://cloudquery.io/docs/plugins/sources/aws/configuration](https://cloudquery.io/docs/plugins/sources/aws/configuration)
```

**Review by erezrokah — APPROVED** [review]
Let's merge this and follow up in future PRs if we find more issues

**Pull request merged**
